src/resources/assets/assetUtils.ts

```diff
@@ -103,18 +103,17 @@ export async function promiseEachInSequence<RequestType, ResponseType>(
   inputs: RequestType[],
   promiser: (input: RequestType) => Promise<ResponseType>
 ) {
-  return inputs.reduce((previousPromise, input, index) => {
-    return previousPromise.then(result => {
-      return promiser(input)
-        .catch(err => {
-          return Promise.reject({
-            errors: [err],
-            failed: inputs.slice(index),
-            succeded: inputs.slice(0, index),
-            responses: result,
-          });
-        })
-        .then(Array.prototype.concat.bind(result));
-    });
-  }, Promise.resolve(new Array()));
+  return inputs.reduce(async (previousPromise, input, index) => {
```

Comment:
I would imagine we have to `Promise.all` this, no?

`return Promise.all(inputs.reduce...)`

Reply:
well no
`Promise.all` resolves all promises simultaneously, while we need to resolve them in sequence here.

Reply:
No, reduce returns one element

Reply:
which will look something like `Promise1().then(Promise2.then(Promise3.then))`

Reply:
True, but then we could return an array of promises. Nevermind -- if this works then crack on

Reply:
no no, the array won't work.
we need to send requests one by one in here :)